jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/senders/HttpSender.java

```diff
@@ -119,13 +134,68 @@ public Builder withAuth(String authToken) {
       return this;
     }
 
+    public Builder withCertificatePinning(String[] sha256certs /* comma separated */) {
+      pins.addAll(Arrays.asList(sha256certs));
+      return this;
+    }
+
+    /**
+     * Enable accepting self-signed certificates. This will only take effect if pinning is provided.
+     */
+    public Builder acceptSelfSigned() {
+      this.selfSigned = true;
+      return this;
+    }
+
     public HttpSender build() {
       if (authInterceptor != null) {
         clientBuilder.addInterceptor(authInterceptor);
       }
+      try {
+        // Just obtain hostname for SSL feature. Failure is OK if plain http is used.
+        final URI uri = new URI(endpoint);
+        String hostname = hostname = uri.getHost();
+
+        if (uri.getScheme() != null && uri.getScheme().equals("https")) {
```

Comment:
You can avoid the null check if you use this: 
```java
"https".equals(uri.getScheme())
```

Reply:
Thanks!